tests/ext/httplib/test_httplib.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import requests
+
+from aws_xray_sdk.core import patch
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+
+
+patch(('httplib',))
+
+# httpbin.org is created by the same author of requests to make testing http easy.
+BASE_URL = 'httpbin.org'
+
+
+@pytest.fixture(autouse=True)
+def construct_ctx():
+    """
+    Clean up context storage on each test run and begin a segment
+    so that later subsegment can be attached. After each test run
+    it cleans up context storage again.
+    """
+    xray_recorder.configure(service='test', sampling=False, context=Context())
+    xray_recorder.clear_trace_entities()
+    xray_recorder.begin_segment('name')
+    yield
+    xray_recorder.clear_trace_entities()
+
+
+def test_ok():
```

Comment:
Not sure your intention on using `requests` to test `httplib`. This unit test is valid given the knowledge of `requests` depends on `httplib`. But I feel like this is a little bit anti-pattern since the scope is beyond `httplib` and involves `requests` implementation.

Reply:
rewrote to use httplib